### ISSUE TITLE
Getting id for android annotation

### ISSUE THREAD

**Issue by maqen** [nitaliano/react-native-mapbox-gl#294]
Hi!

Is the id passed to the annotation object upon adding it to a map?

Logging seems to only return

```
{ src: 
  {
    longitude: -73.9878,
    subtitle: 'Neat, this is a custom annotation image',
    latitude: 40.7223,
    title: 'Important!' 
  }
}
```

Adding it to state as per example

```
annotations: [{
  coordinates: [40.7223, -73.9878],
  type: 'point',
  title: 'Important!',
  subtitle: 'Neat, this is a custom annotation image',
  id: 'marker2'
}]
```

Has this something to do with `id` never being set on the native annotation? 
https://github.com/mapbox/react-native-mapbox-gl/blob/master/android/src/main/java/com/mapbox/reactnativemapboxgl/ReactNativeMapboxGLManager.java#L126

As opposed to being set in iOS like this https://github.com/mapbox/react-native-mapbox-gl/blob/master/ios/RCTMapboxGL/RCTMapboxGLManager.m#L385

**Comment by vespakoen** [comment]
Hey!

The id / setId methods are used by mapbox internally and cannot be used for that purpose.
However, this is a much needed feature, right now I am abusing the subtitle to know the id of the marker that was clicked, which sucks.
I think the javascript side can use the "id" property, then stuff it somewhere in the mapbox gl marker on the native side. and map it back to the id field in the javascript context.

I will see if I can make a pull request for this.

Until that time, here is my +1 for this issue as well.

By the way, your title suggests this already works on IOS, is that the case?

**Comment by maqen** [comment]
Yes @vespakoen it works on iOS, please see https://github.com/mapbox/react-native-mapbox-gl/blob/master/ios/RCTMapboxGL/RCTMapboxGLManager.m#L374

Let's look forward to some way to expose/introduce id for android native SDK

**Comment by vespakoen** [comment]
https://github.com/mapbox/mapbox-gl-native/blob/master/platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/annotations/Annotation.java#L34

**Comment by vespakoen** [comment]
https://github.com/mapbox/mapbox-gl-native/issues/3756

**Comment by vespakoen** [comment]
Basically, we have to wait for 4.0.0 support to land (there is already a pull request for that I think).

**Comment by jayfunk** [comment]
#313 is the PR.

**Comment by dapetcu21** [comment]
This is fixed in #365
